apps/freeablo/faworld/position.h

```diff
@@ -33,16 +33,33 @@ namespace FAWorld
         Misc::Direction getDirection() const { return mDirection; }
         void setDirection(Misc::Direction mDirection);
 
-        bool isMoving() const { return mMoving; }
-        int32_t getDist() const { return mDist; }
+        bool isMoving() const { return mMovementType != MovementType::stopped; }
+        Misc::Point getFractionalPos() const { return mFractionalPos; }
 
-        void stop();
-        void start();
+        int32_t getSpeed() const { return mSpeed; }
+        void setSpeed(int32_t speed) { mSpeed = speed; }
+
+        void stopMoving();
+        void moveToPoint(const Misc::Point& dest);
+        void moveInDirection();
 
     private:
+        enum class MovementType
+        {
+            stopped,
+            // Chebyshev movement with a destination tile.
+            // Movement to any neighboring tile takes the same time.
+            towardPointChebyshev,
+            // Euclidean movement in any direction.
+            // Normal/Pythagorean directional movement.
+            inDirectionEuclidean
```

Comment:
I would PascalCase these enum members

Reply:
Also, I get that chebyshev is the proper mathematical name for this movement type, but in this context I think it might be better to call it GridLocked vs FreeMovement or something like that.